BGAnimations/ScreenSelectMusic overlay/SongSearch/CandidateItemMT.lua

```diff
@@ -0,0 +1,78 @@
+local paneHeight = 319
+local paneWidth = 319
+
+-- the metatable for an item in the song search results.
+return {
+	__index = {
+		create_actors = function(self, name)
+			self.name=name
+
+			local af = Def.ActorFrame{
+				Name=name,
+
+				InitCommand=function(subself)
+					self.container = subself
+					subself:MaskDest()
+					subself:diffusealpha(0)
+				end,
+			}
+
+			-- Song Name
+			af[#af+1] = Def.BitmapText{
+				Font="Common Normal",
+				Name="Song",
+				InitCommand=function(subself)
+					self.song_name = subself
+					subself:y(0):diffusealpha(0):maxwidth(155)
+				end,
+				OnCommand=function(subself)
+					subself:sleep(0.13):linear(0.05):diffusealpha(1)
+				end
+			}
+
+			return af
+		end,
+
+		transform = function(self, item_index, num_items, has_focus)
+			self.container:finishtweening()
+	
+			if has_focus then
+				self.container:accelerate(0.1)
+				self.container:diffuse( GetCurrentColor() )
+				self.container:glow(color("1,1,1,0.5"))
+				MESSAGEMAN:Broadcast("UpdateSearchResult", {songOrExit=self.song_name.songOrExit})
+			else
+				self.container:glow(color("1,1,1,0"))
+				self.container:accelerate(0.1)
+				self.container:diffuse(color("#888888"))
+				self.container:glow(color("1,1,1,0"))
+			end
+
+			self.container:y(30 * item_index)
+
+			if item_index <= 1 or item_index >= num_items then
+				self.container:diffusealpha(0)
+			else
+				self.container:diffusealpha(1)
+			end
+
+			self.container:diffusealpha(1)
+
+		end,
+
+		set = function(self, info)
+			if info == nil then self.song_name:settext("") return end
+
+			local songOrExit = info.songOrExit
+
+			-- We need some way to differentiate between Songs and the "Exit" text.
```

Comment:
In terms of UI, it would help to set the exit row's font to `"Common Bold"`.   This would more distinctly set it apart and not rely solely on color.

This will probably require you to have an additional BitmapText actor in the scroller item: one for those that will use `Common Normal` and another in case it needs `Common Bold`.  You could set each BMT's `visible()` based on the presence of text, or similar.

With that in place, you could diffuse to `GetCurrentColor()` rather than red, but this is more aesthetic than practical. :)

Reply:
The comment was originally used to explain why we're checking for "type(string)" but I do like using Common Bold as well. I'm not sure about using `GetCurrentColor()` thought because that's what the highlight for the text uses and it blends in.